daweb/js1/cykly/cvlekce/pole-v-divadle.md

```diff
@@ -1,11 +1,41 @@
 ---
 title: Pole v divadle
 demand: 1
+solutionAccess: lock
 ---
 
-Celé toto cvičení prováďejte v JavaScript konzoli.
+Celé toto cvičení provádějte v JavaScript konzoli.
 
-1. Vytvořte nějaké pole celých čísel, například počty diváků na několika po sobě jdoucích divadelních představeních. Použije metodu `push` a přidejte do pole počet diváků na novém představení.
-1. Vytvořte nějaké pole desetinných čísel, například zaplněnost divadla v několika po sobě jdoucích představeních.
-1. Vytvořte nějaké pole řetězců, například seznam názvů několika divadelních představení, která divadlo hraje. Uložte toto pole do proměnné `plays`. Uložte do nějaké proměnné druhou položku tohoto pole. Pomocí metody `shift` odstraňte z pole první představení, které už divadlo nehraje.
-1. Do proměnné `reviews` uložte pole hodnocení, které obdržela divadelní hra :i[Plyšáci na útěku] v různých recenzních časopisech. Hodnocení je vždy dvouprvkové pole obsahující název recenzního časopisu jako řetězec a samotné hodnocení jako číslo mezi 1 až 10. Pomocí metody `unshift` přidejte na začátek nové hodnocení z časopisu :i[Divadelní oběžník].
+1. Vytvořte nějaké pole celých čísel, například počty diváků na několika po sobě jdoucích divadelních představeních. Použijte metodu `push` a přidejte do pole počet diváků na novém představení.
+1. Vytvořte nějaké pole desetinných čísel, například zaplněnost divadla v několika po sobě jdoucích představeních. Poté pomocí metody `pop` odstraňte poslední hodnotu v poli.
+1. Vytvořte nějaké pole řetězců, například seznam názvů několika divadelních představení, která divadlo hraje. Uložte toto pole do proměnné `plays`. Uložte do nějaké proměnné druhou položku tohoto pole.
+1. Uložte do proměnné `reviews` pole hodnocení, které obdržela divadelní hra :i[Plyšáci na útěku] v různých recenzních časopisech. Hodnocení je vždy dvouprvkové pole obsahující název recenzního časopisu jako řetězec a samotné hodnocení jako číslo mezi 1 až 10. Pomocí metody `includes` zjistěte, jestli je v poli hodnocení od časopisu :i[Divadelní Luk].
```

Comment:
Tohle cvičení se mi nikdy nelíbilo. Z logiky věci by ten vnitřní datový typ měl být objekt s property třeba `nazev` a `hodnoceni`, nemělo by to být pole. Pokud chceme zachovat to prostředí divadelních her, tak co třeba pole s řadami sedadel a každá řada bude pole, kde budou hodnoty `true`/`false` podle toho, zda je sedadlo obsazené? A stačí nějakou komorní scénu, kde bude pár sedadel :-)

Reply:
Souhlasím, ale nechávám na Martinovi :)

Reply:
Taky sohlasím, opravím to.